addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -214,13 +214,14 @@ def unlink(self):
         self.sudo().rtc_session_ids.unlink()  # ensure unlink overrides are applied
         # Always unlink members of sub-channels as well. Member list should be
         # kept in sync.
-        sub_members = []
-        for parent in self.channel_id.filtered(lambda c: c.sub_channel_ids):
-            sub_members += parent.sub_channel_ids.channel_member_ids.filtered(
-                lambda m: m.partner_id in set(parent.channel_member_ids.partner_id)
-                if m.partner_id
-                else m.guest_id in set(parent.channel_member_ids.guest_id)
-            )
+        sub_members = self.env["discuss.channel.member"].search(
+            [
+                "|",
+                ("partner_id", "in", self.partner_id.ids),
+                ("guest_id", "in", self.guest_id.ids),
+                ("channel_id", "in", self.channel_id.sub_channel_ids.ids),
```

Comment:
This is not correct. Each channel should be mapped to specific guests or partners.

If unlink is called on partner 1 of channel 2 and partner 2 of channel 1, this will incorrectly also return partner 1 of channel 2 and partner 2 of channel 2.

Reply:
Done